src/index.js

```diff
@@ -278,14 +281,12 @@ export default function undoable (reducer, rawConfig = {}) {
           return config.history
         }
 
-        if (config.filter && typeof config.filter === 'function') {
-          if (!config.filter(action, res, history.present)) {
-            debug('filter prevented action, not storing it')
-            debugEnd()
-            return {
-              ...history,
-              present: res
-            }
+        if (typeof config.filter === 'function' && !config.filter(action, res, history.present)) {
+          debug('filter prevented action, not storing it')
+          debugEnd()
+          return {
+            ...history,
+            present: res
           }
```

Comment:
Don't worry about performance issues like that, it's a very very small gain :wink:

I think it was just done because I'm used to checking if the property exists before working with it.